kiota_http/middleware/options/redirect_handler_option.py

```diff
@@ -25,8 +25,10 @@ def __init__(
             raise ValueError(
                 "MaxLimitExceeded. Negative value for max_redirect property is invalid"
             )
+        else:
+            self._max_redirect = self.DEFAULT_MAX_REDIRECT
```

Comment:
This means that if `max_redirect` is passed as valid value (e.g `7`), the value will not be respected/used and it will be set to `self.DEFAULT_MAX_REDIRECT` either way.